### DISCUSSION TITLE
Is blend shape supported in AssImp for FBX files

### DISCUSSION THREAD

**Discussion by meetjatin88** [assimp/assimp#3658] [Q&A] [answered]
Greetings, i have an fbx file that has blend shape animation and reading the code i am not certain whether blend shape animation is supported in AssImp or not, for eg  If you check https://github.com/assimp/assimp/blob/master/include/assimp/mesh.h#L0707 then from the comments it seems like it is only supported for collada loader
 but if you check https://github.com/assimp/assimp/blob/3929ea064468dfe54b0d2de7e45aa4702d5a8f12/code/FBX/FBXConverter.cpp#L1164 you can see that aiAnimMesh is getting populated from blendshapes found in the fbx file.

Also if i look at the latest docs at https://assimp-docs.readthedocs.io/en/latest/usage/use_the_lib.html?highlight=blend#pseudo-code-listing it states
 
Blenshapes
ToDo!

Therefore would like to know for sure whether blend shape is supported also if blend shape is supported then once i have the aiAnimMesh data do i need to create an aiMesh object from it ( if so is there a helper method for that, since i am assuming data in aiAnimMesh like verts etc must be delta's from the base mesh). Would be great if someone or @kimkulling can point me to some sample code as to how to get the shape datas of fbx's blendshapes.

**Comment by kimkulling** [comment, marked as answer]
On the current master: yes.
